contracts/Core.sol

```diff
@@ -22,36 +22,58 @@ pragma solidity ^0.4.23;
 // ----------------------------------------------------------------------------
 
 import "./CoreInterface.sol";
+import "./MerklePatriciaProof.sol";
+import "./util.sol";
+import "./WorkersInterface.sol";
 
 
 /// @dev Core is a minimal stub that will become the anchoring and consensus point for
 ///      the utility chain to validate itself against
-contract Core is CoreInterface {
+// TODO - truffle test cases
+contract Core is CoreInterface, Util {
 
 	/*
-	 *  Structures
-	 */
-	// struct stakeTokenTuple {
-	// 	address stake;
-	// 	address token;
-	// }
+    *    Events
+    */
+	event StateRootCommitted(uint256 blockHeight, bytes32 stateRoot);
 
+	event AccountProved(uint256 blockHeight, bytes32 stateRoot, bytes encodedAddress, bytes32 storageRoot);
 
 	/*
 	 *  Storage
 	 */
-	/// registrar registers for the two chains
-	address private coreRegistrar;
+	/** Mapping of block height to state root of the block.  */
+	mapping (uint /* block height */ => bytes32) private stateRoots;
+	/** Mapping of block height to storafe root of the block.  */
+	mapping (uint /* block height */ => bytes32) private storageRoots;
+
+	// TODO - Do we need below mapping?
+	// mapping(bytes32 => address) stakeTokenTuple;
+
 	/// chainIdOrigin stores the chainId this chain
 	uint256 private coreChainIdOrigin;
 	/// chainIdRemote stores the chainId of the remote chain
 	uint256 private coreChainIdRemote;
 	/// OpenST remote is the address of the OpenST contract
 	/// on the remote chain
 	address private coreOpenSTRemote;
-	// /// 
-	// mapping(bytes32 => address) stakeTokenTuple;
+	/// registrar registers for the two chains
+	address private coreRegistrar;
+    /// Latest block height of block which state root was committed.
+    uint256 private latestStateRootBlockHeight;
+    /// Latest block height of block which storage root was committed.
+    uint256 private latestStorageRootBlockHeight;
```

Comment:
Was wondering, can ``` stateRoots ``` , ```  storageRoots ``` , ``` latestStateRootBlockHeight ``` and ``` latestStorageRootBlockHeight ``` be public ? So we may get rid of the getter functions ? 
Do we have any advantage to keep variables private and expose the getter functions.

Reply:
Only advantage is functions have more capability and can have logic if needed.